klog_file.go

```diff
@@ -44,12 +44,44 @@ func createLogDirs() {
 }
 
 var (
-	pid      = os.Getpid()
-	program  = filepath.Base(os.Args[0])
-	host     = "unknownhost"
-	userName = "unknownuser"
+	pid          = os.Getpid()
+	program      = filepath.Base(os.Args[0])
+	host         = "unknownhost"
+	userName     = "unknownuser"
+	userNameOnce sync.Once
 )
 
+func getUserName() string {
+	userNameOnce.Do(func() {
+		// On Windows, the Go 'user' package requires netapi32.dll.
+		// This affects Windows Nano Server:
+		//   https://github.com/golang/go/issues/21867
+		// Fallback to using environment variables.
+		if runtime.GOOS == "windows" {
+			u := os.Getenv("USERNAME")
+			if len(u) == 0 {
+				return
+			}
+			// Sanitize the USERNAME since it may contain filepath separators.
+			u = strings.Replace(u, `\`, "_", -1)
+
+			// user.Current().Username normally produces something like 'USERDOMAIN\USERNAME'
+			d := os.Getenv("USERDOMAIN")
+			if len(d) != 0 {
+				userName = d + "_" + u
+			} else {
+				userName = u
+			}
+		} else {
+			current, err := user.Current()
+			if err == nil {
+				userName = current.Username
+			}
+		}
+	})
+	return userName
+}
+
 func init() {
```

Comment:
hi, the addition of getUserName() seems fine, but why keep the logic in init() though?

Reply:
🤦‍♂️🤦‍♂️ because I was checking something and pressed "undo" not paying attention to "redo"

Give me 5; heading to my keyboard and will fix

Reply:
Fixed! Sorry for that


![nothing-so-see-here](https://user-images.githubusercontent.com/1804568/81509132-d7eee300-9308-11ea-8d95-ff3d296f547e.gif)